cypress/integration/page.spec.js

```diff
@@ -1,148 +1,1259 @@
+          .should('have.text', 'harum non quasi et ratione\ntempore iure ex voluptates in ratione\nharum architecto fugit inventore cupiditate\nvoluptates magni quo et');
+      });
 
-    cy.getByDataCy('postDetails')
-      .should('contain', commentBody);
+      it('should disappear after selecting another user', () => {
+        page.mockPost1Comments();
+        page.postButton(0).click();
 
-    cy.contains('li', commentBody)
-      .find('button')
-      .contains('X')
-      .click();
+        page.mockUser2Posts();
+        userSelector.select(1);
 
-    cy.wait('@deleteComment')
-      .its('response.statusCode')
-      .should('eq', 200);
+        postDetails.el().should('not.exist');
+      });
+    });
 
-    cy.getByDataCy('postDetails')
-      .should('not.contain', commentBody);
+    describe('after comments loading error', () => {
+      beforeEach(() => {
+        page.mockError('**/comments?postId=1', 'post1CommentsRequest');
+        page.postButton(0).click();
+        page.waitForRequest('@post1CommentsRequest');
+      });
+
+      it('should hide comments loader', () => {
+        postDetails.commentsLoader().should('not.exist');
+      });
+
+      it('should show comments error', () => {
+        postDetails.commentsError().should('exist');
+      });
+
+      it('should not show no comments message', () => {
+        postDetails.noCommentsMessage().should('not.exist');
+      });
+
+      it('should not show WriteCommentButton', () => {
+        postDetails.writeCommentButton().should('not.exist');
+      });
+
+      it('should not show NewCommentForm', () => {
+        newCommentForm.el().should('not.exist')
+      });
+    });
+
+    describe('after empty comments received', () => {
+      beforeEach(() => {
+        cy.intercept('**/comments?postId=1', { body: [] }).as('post1CommentsRequest');
+        page.postButton(0).click();
+        page.waitForRequest('@post1CommentsRequest');
+      });
+
+      it('should show no comments message', () => {
+        postDetails.noCommentsMessage().should('exist');
+      });
+
+      it('should hide comments loader', () => {
+        postDetails.commentsLoader().should('not.exist');
+      });
+
+      it('should not show comments error', () => {
+        postDetails.commentsError().should('not.exist');
+      });
+
+      it('should show WriteCommentButton', () => {
+        postDetails.writeCommentButton().should('exist');
+      });
+
+      it('should not show NewCommentForm', () => {
+        newCommentForm.el().should('not.exist')
+      });
+    });
+
+    describe('after selecting another post', () => {
+      beforeEach(() => {
+        page.mockPost1Comments();
+        page.postButton(0).click();
+        page.waitForRequest('@post1CommentsRequest');
+      });
+
+      describe('', () => {
+        beforeEach(() => {
+          cy.clock();
+          page.spyOn('**/comments?postId=2', 'post2Coments', { fixture: 'post2Comments' });
```

Comment:
There is a typo in the alias name: 'post2Coments'. It should be 'post2Comments'.